cli/sawtooth_cli/config.py

```diff
@@ -75,41 +106,87 @@ def _do_config_set(args):
     """
     settings = [s.split('=', 1) for s in args.setting]
 
-    with open(args.key, 'r') as key_file:
-        wif_key = key_file.read().strip()
-        signing_key = signing.encode_privkey(
-            signing.decode_privkey(wif_key, 'wif'), 'hex')
-        pubkey = signing.encode_pubkey(
-            signing.generate_pubkey(signing_key), 'hex')
+    pubkey, signing_key = _read_signing_keys(args.key)
 
-    txns = [_create_config_txn(pubkey, signing_key, setting)
+    txns = [_create_propose_txn(pubkey, signing_key, setting)
             for setting in settings]
-    txn_ids = [txn.header_signature for txn in txns]
 
-    batch_header = BatchHeader(signer_pubkey=pubkey,
-                               transaction_ids=txn_ids).SerializeToString()
-
-    batch = Batch(
-        header=batch_header,
-        header_signature=signing.sign(batch_header, signing_key),
-        transactions=txns
-    )
+    batch = _create_batch(pubkey, signing_key, txns)
 
     batch_list = BatchList(batches=[batch]).SerializeToString()
 
     try:
         with open(args.output, 'wb') as batch_file:
             batch_file.write(batch_list)
-    except:
-        raise CliException('Unable to write to {}'.format(args.output))
+    except IOError as ioe:
+        raise CliException(
+            'Unable to write to {}: {}'.format(args.output, ioe))
+
+
+def _do_config_propose(args):
+    """Executes the 'propose subcommand. Given a key file, a URL and a setting
+    key and value, it generates a ConfigPayload.PROPOSE sawtooth_config
+    transaction and submits it via the validator REST api.
+    """
+    pubkey, signing_key = _read_signing_keys(args.key)
+    txn = _create_propose_txn(pubkey, signing_key, (args.setting, args.value))
+    batch = _create_batch(pubkey, signing_key, [txn])
+    batch_list = BatchList(batches=[batch])
+
+    rest_client = RestClient(args.url)
+    rest_client.send_batches(batch_list)
+
+
+def _read_signing_keys(key_filename):
+    """Reads the given file as a WIF formatted key.
+
+    Args:
+        key_filename: The filename where the key is stored. If None,
+            defaults to the default key for the current user.
+
+    Returns:
+        tuple (str, str): the public and private key pair
+
+    Raises:
+        CliException: If unable to read the file.
+    """
+    filename = key_filename
+    if filename is None:
+        filename = os.path.join(os.path.expanduser('~'),
+                                '.sawtooth',
+                                'keys',
+                                getpass.getuser() + '.wif')
```

Comment:
Does there need to be some centrally-located function that defaults the location of sawtooth configuration files so that every utility does not have to construct the path like this?

For example, when I fire up a validator, if not provided with a key, it assumes this is the default:

~/sawtooth/etc/keys/validator.wif

If I run 'sawtooth keygen' it defaults to the above.  Seems to me that there should be one place, and one place only, that defines the default key file name.